### PR TITLE
feat: add preview for review in demo flow

### PR DESCRIPTION
**Pull request by cavacado**
## Summary

What is the background of this pull request?
This pr brings the preview functionality from doc-creator to tt-website
one can consider this a sloppy first draft

## Changes

- What are the changes made in this pull request?
- refactor the way we are getting wrapped document
- document is constructed in the component level before it is passed on to the sagas, so we dont need to compute twice as there is a requirement from the renderer side to have a raw document

## TODOS:
- bring toggle component to tt-ui-components
- add tests for makeRawDoc?? maybe

### CODE REVIEW

File: src/components/Demo/DemoCreate/DemoCreateReview/DemoPreview.tsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
+import React, { useState, useCallback, useRef, FunctionComponent } from "react";
+import {
+  FrameConnector,
+  HostActions,
+  renderDocument,
+  FrameActions,
+} from "@govtechsg/decentralized-renderer-react-components";
+import { OpenAttestationDocument, utils } from "@govtechsg/open-attestation";
+
+type Dispatch = (action: HostActions) => void;
```
Comment (simboonlong) [inline src/components/Demo/DemoCreate/DemoCreateReview/DemoPreview.tsx]: move to types? was used in decentralise renderer as well

File: src/components/Demo/DemoCreate/DemoCreateReview/DemoPreview.tsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+interface DocumentPreview {
+  document: OpenAttestationDocument;
+}
+
+export const DocumentPreview: FunctionComponent<DocumentPreview> = ({ document }) => {
```
Comment (simboonlong) [inline src/components/Demo/DemoCreate/DemoCreateReview/DemoPreview.tsx]: think long ago had a discussion on whether to name interface exact same name as component, not sure about this wdyt

File: src/components/Demo/DemoCreate/DemoCreateReview/Toggle.tsx [hunk 3]
```diff
@@ -0,0 +1,34 @@
+// TODO move to tt-ui-components
+import React, { FunctionComponent } from "react";
```
Comment (simboonlong) [inline src/components/Demo/DemoCreate/DemoCreateReview/Toggle.tsx]: shall we do that

File: src/components/Demo/DemoCreate/DemoCreateReview/DemoPreview.tsx (hunk 2 above)
Comment (cavacado) [reply src/components/Demo/DemoCreate/DemoCreateReview/DemoPreview.tsx]: can, will rename the props interface accordingly

File: src/components/Demo/DemoCreate/DemoCreateReview/Toggle.tsx (hunk 3 above)
Comment (cavacado) [reply src/components/Demo/DemoCreate/DemoCreateReview/Toggle.tsx]: yup

File: src/components/Demo/DemoCreate/DemoCreateReview/DemoPreview.tsx (hunk 1 above)
Comment (cavacado) [reply src/components/Demo/DemoCreate/DemoCreateReview/DemoPreview.tsx]: 👌

**Review by simboonlong — APPROVED** [review]
lgtm

**Pull request merged**